rocm-related: 5.3.1 -> 5.3.3 & misc changes

Madouura opened the pull request:
###### Description of changes
Tracking: #197885

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

Madouura committed:
rocm-related: add update script

rocprim: add update script

hipcub: add update script

rocsparse: add update script

hipsparse: add update script

rccl: add update script

rocfft: add update script

tensile: add update script

rocblas: add update script

miopengemm: add update script

clang-ocl: add update script

rocthrust: add update script

rocmlir: add update script

Madouura committed:
rocm-related: ignore same hash

rocclr: ignore same hash

rocm-comgr: ignore same hash

rocm-device-libs: ignore same hash

rocm-opencl-runtime: ignore same hash

rocm-runtime: ignore same hash

rocm-thunk: ignore same hash

rocm-cmake: ignore same hash

rocminfo: ignore same hash

rocm-smi: ignore same hash

Madouura committed:
rocm-related: 5.3.1 → 5.3.3

rocm-smi: 5.3.1 → 5.3.3

rocm-cmake: 5.3.1 → 5.3.3

rocm-thunk: 5.3.1 → 5.3.3

rocm-runtime: 5.3.1 → 5.3.3

rocm-opencl-runtime: 5.3.1 → 5.3.3

rocm-device-libs: 5.3.1 → 5.3.3

rocm-comgr: 5.3.1 → 5.3.3

rocclr: 5.3.1 → 5.3.3

rocminfo: 5.3.1 → 5.3.3

llvmPackages_rocm.llvm: 5.3.1 → 5.3.3

hip: 5.3.1 → 5.3.3

hipcub: 2.12.0-5.3.1 → 2.12.0-5.3.3

rocsparse: 2.3.2-5.3.1 → 2.3.2-5.3.3

hipsparse: 2.3.1-5.3.1 → 2.3.1-5.3.3

rccl: 2.12.10-5.3.1 → 2.12.10-5.3.3

rocfft: 1.0.18-5.3.1 → 1.0.18-5.3.3

tensile: 4.34.0-5.3.1 → 4.34.0-5.3.3

rocblas: 2.45.0-5.3.1 → 2.45.0-5.3.3

miopengemm: 5.3.1 → 5.3.3

clang-ocl: 5.3.1 → 5.3.3

rocthrust: 2.16.0-5.3.1 → 2.16.0-5.3.3

rocprim: 2.11.0-5.3.1 → 2.11.1-5.3.3

rocmlir: 5.3.1 → 5.3.3

Madouura commented:
Looks like I forgot ``composable_kernel`` and ``rocmlir`` - whoops.

Madouura committed:
composable_kernel: unstable-2022-11-02 -> unstable-2022-11-19

Madouura commented:
Result of `nixpkgs-review pr 202373` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>28 packages built:</summary>
  <ul>
    <li>blender-hip</li>
    <li>clang-ocl</li>
    <li>composable_kernel</li>
    <li>hip</li>
    <li>hipcub</li>
    <li>hipsparse</li>
    <li>llvmPackages_rocm.clang</li>
    <li>llvmPackages_rocm.clangNoCompilerRt</li>
    <li>llvmPackages_rocm.llvm</li>
    <li>miopengemm</li>
    <li>rccl</li>
    <li>rocblas</li>
    <li>rocclr</li>
    <li>rocfft</li>
    <li>rocm-cmake</li>
    <li>rocm-comgr</li>
    <li>rocm-device-libs</li>
    <li>rocm-opencl-icd</li>
    <li>rocm-opencl-runtime</li>
    <li>rocm-runtime</li>
    <li>rocm-smi</li>
    <li>rocm-thunk</li>
    <li>rocminfo</li>
    <li>rocmlir</li>
    <li>rocprim</li>
    <li>rocsparse</li>
    <li>rocthrust</li>
    <li>tensile</li>
  </ul>
</details>

Flakebi reviewed (approved):
Nice!
I tried some OpenCL and ROCm examples and they worked fine.

I found a way to call the update script with pressing enter only a single time: `nix-shell maintainers/scripts/update.nix --argstr commit true --argstr keep-going true --arg predicate '(path: pkg: builtins.elem (pkg.pname or null) [ "clang-ocl" "composable_kernel" "hip" "hipcub" "hipsparse" "miopengemm" "rccl" "rocblas" "rocclr" "rocfft" "rocprim" "rocsparse" "rocthrust" "rocm-cmake" "rocm-comgr" "rocm-device-libs" "rocm-llvm" "rocm-opencl-runtime" "rocm-runtime" "rocm-smi" "rocm-thunk" "rocminfo" "tensile" ])'`

Madouura commented:
Niiiiice. :+1:

Madouura commented:
In my TODO I have:
- Change all relevant derivations from ``rec`` to ``finalAttrs``
- Change all relevant derivation's maintainers to rocm team + previous non-team maintainers
Should I include them in this PR? And if I do, should I roll all of these changes into one commit per derivation?
On one hand the flexibility of just being able to revert each commit if needed is nice, on the other hand it feels like commit spam.

acowley commented:
I think one commit per listed change to the package set, not one commit per derivation, is fine.

Madouura committed:
rocm-related: change maintainers to rocm team

rocprim: change maintainers to rocm team

hipcub: change maintainers to rocm team

rocsparse: change maintainers to rocm team

hipsparse: change maintainers to rocm team

rccl: change maintainers to rocm team

rocfft: change maintainers to rocm team

tensile: change maintainers to rocm team

rocblas: change maintainers to rocm team

miopengemm: change maintainers to rocm team

clang-ocl: change maintainers to rocm team

rocthrust: change maintainers to rocm team

rocm-cmake: change maintainers to rocm team

composable_kernel: change maintainers to rocm team

rocmlir: change maintainers to rocm team

Madouura committed:
rocm-related: change maintainers to rocm team + old

rocclr: change maintainers to rocm team + old

rocm-comgr: change maintainers to rocm team + old

rocm-device-libs: change maintainers to rocm team + old

rocm-opencl-runtime: change maintainers to rocm team + old

rocm-runtime: change maintainers to rocm team + old

rocm-thunk: change maintainers to rocm team + old

rocminfo: change maintainers to rocm team + old

rocm-smi: change maintainers to rocm team + old

rocm-llvm: change maintainers to rocm team + old

hip: change maintainers to rocm team + old

Madouura committed:
rocm-related: use finalAttrs

hip: use finalAttrs

rocm-llvm: use finalAttrs

clang-ocl: use finalAttrs

composable_kernel: use finalAttrs

hipcub: use finalAttrs

hipsparse: use finalAttrs

miopengemm: use finalAttrs

rccl: use finalAttrs

rocblas: use finalAttrs

rocclr: use finalAttrs

rocfft: use finalAttrs

rocm-comgr: use finalAttrs

rocm-device-libs: use finalAttrs

rocm-opencl-runtime: use finalAttrs

rocm-runtime: use finalAttrs

rocm-thunk: use finalAttrs

rocmlir: use finalAttrs

rocprim: use finalAttrs

rocsparse: use finalAttrs

rocthrust: use finalAttrs

rocm-cmake: use finalAttrs

rocminfo: use finalAttrs

rocm-smi: use finalAttrs

Madouura commented:
Not using ``finalAttrs`` for ``tensile``  because it doesn't work when used as an optional ``null`` for another derivation input.

Madouura commented:
Result of `nixpkgs-review pr 202373` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>27 packages built:</summary>
  <ul>
    <li>blender-hip</li>
    <li>clang-ocl</li>
    <li>composable_kernel</li>
    <li>hip</li>
    <li>hipcub</li>
    <li>hipsparse</li>
    <li>llvmPackages_rocm.clang</li>
    <li>llvmPackages_rocm.clangNoCompilerRt</li>
    <li>llvmPackages_rocm.llvm</li>
    <li>miopengemm</li>
    <li>rccl</li>
    <li>rocblas</li>
    <li>rocclr</li>
    <li>rocm-cmake</li>
    <li>rocm-comgr</li>
    <li>rocm-device-libs</li>
    <li>rocm-opencl-icd</li>
    <li>rocm-opencl-runtime</li>
    <li>rocm-runtime</li>
    <li>rocm-smi</li>
    <li>rocm-thunk</li>
    <li>rocminfo</li>
    <li>rocmlir</li>
    <li>rocprim</li>
    <li>rocsparse</li>
    <li>rocthrust</li>
    <li>tensile</li>
  </ul>
</details>

Madouura commented:
Only change was ``rocmlir`` using the now-added tags. https://github.com/ROCmSoftwarePlatform/rocMLIR/tags

mweinelt commented on pkgs/development/compilers/llvm/rocm/default.nix:
```diff
@@ -1,7 +1,7 @@
 
 let
-  version = "5.3.1";
+  version = "5.3.3";
```
version changed, but hash stayed the same?

Flakebi replied:
Yes, this happens quite often with ROCm. All projects get a new tag for a minor version, but only some of them have actual changes.

mweinelt commented on pkgs/development/libraries/clang-ocl/default.nix:
```diff
@@ -1,21 +1,22 @@
     repo = "clang-ocl";
-    rev = "rocm-${rocmVersion}";
+    rev = "rocm-${finalAttrs.rocmVersion}";
     hash = "sha256-uMSvcVJj+me2E+7FsXZ4l4hTcK6uKEegXpkHGcuist0=";
```
version change, but no hash change?

mweinelt commented:
Result of `nixpkgs-review pr 202373` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>27 packages built:</summary>
  <ul>
    <li>blender-hip</li>
    <li>clang-ocl</li>
    <li>composable_kernel</li>
    <li>hip</li>
    <li>hipcub</li>
    <li>hipsparse</li>
    <li>llvmPackages_rocm.clang</li>
    <li>llvmPackages_rocm.clangNoCompilerRt</li>
    <li>llvmPackages_rocm.llvm</li>
    <li>miopengemm</li>
    <li>rccl</li>
    <li>rocblas</li>
    <li>rocclr</li>
    <li>rocm-cmake</li>
    <li>rocm-comgr</li>
    <li>rocm-device-libs</li>
    <li>rocm-opencl-icd</li>
    <li>rocm-opencl-runtime</li>
    <li>rocm-runtime</li>
    <li>rocm-smi</li>
    <li>rocm-thunk</li>
    <li>rocminfo</li>
    <li>rocmlir</li>
    <li>rocprim</li>
    <li>rocsparse</li>
    <li>rocthrust</li>
    <li>tensile</li>
  </ul>
</details>

SuperSandro2000 commented on pkgs/development/libraries/clang-ocl/default.nix:
```diff
@@ -34,11 +35,18 @@ stdenv.mkDerivation rec {
+  passthru.updateScript = writeScript "update.sh" ''
+    #!/usr/bin/env nix-shell
+    #!nix-shell -i bash -p curl jq common-updater-scripts
+    rocmVersion="$(curl -sL "https://api.github.com/repos/RadeonOpenCompute/clang-ocl/releases?per_page=1" | jq '.[0].tag_name | split("-") | .[1]' --raw-output)"
```
This should optionally take a GITHUB_TOKEN to not run into rate limiting

Madouura replied:
Like this? https://github.com/NixOS/nixpkgs/blob/350fd0044447ae8712392c6b212a18bdf2433e71/pkgs/applications/emulators/rpcs3/update.sh

SuperSandro2000 replied:
I was more thinking about something like this https://github.com/NixOS/nixpkgs/blob/b0f8c96b9ef5998f3985c34c90cfd21f1ba8a1ce/pkgs/applications/emulators/citra/update.sh#L30 to make it truly optional

Madouura replied:
Already done in the PR, so we should be good here.
